web/src/containers/DeleteActivity.js

```diff
@@ -0,0 +1,70 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { removeActivity as removeActivityAction } from '../actions/activities';
+import { confirmAlert } from '../components/ConfirmAlert';
+
+const DeleteConfirm = ({ onClose, onConfirm }) => (
+  <div className="p2 sm-p3 bg-white rounded confirm-body">
+    <h2 className="mt0">Please confirm.</h2>
+    <p>Are you sure you’d like to remove this activity?</p>
+    <button
+      className="btn btn-small btn-primary bg-black h6"
+      onClick={() => {
+        onConfirm();
+        onClose();
+      }}
+    >
+      Yes
+    </button>{' '}
+    <button className="btn btn-small btn-outline h6" onClick={onClose}>
+      Cancel
+    </button>
+  </div>
+);
+
+DeleteConfirm.propTypes = {
+  onClose: PropTypes.func.isRequired,
+  onConfirm: PropTypes.func.isRequired
+};
+
+class DeleteActivity extends Component {
+  handleClick = () => {
+    confirmAlert({
+      customUI: ({ onClose }) => (
```

Comment:
One thing that bugs me a little bit about React sometimes is the amount of redirection that happens.  It can make it hard to track where things are actually happening.  This is an example of that.  I don't know if there's a much better way...  Maybe move the `confirmAlert(...` inside the `DeleteConfirm` component?  Then the `DeleteActivity` component wouldn't need to know about `confirmAlert`.  But I don't know that's really useful since they're both in this same module...  🤔

I mean, this is fine, it's just one of those cases where I look at React and realize it's not, in fact, ***perfect***.  😆

Reply:
i agree with you -- this is certainly kinda gross / hard to follow at the least; will think about how to clean up / refine